golem-worker-executor-base/src/durable_host/wasm_rpc/serialized.rs

```diff
@@ -15,11 +15,27 @@
 use crate::durable_host::serialized::SerializableError;
 use crate::services::rpc::RpcError;
 use bincode::{Decode, Encode};
-use golem_wasm_rpc::WitValue;
+use golem_wasm_rpc::{Value, WitValue};
 
 #[derive(Debug, Clone, Encode, Decode)]
 pub enum SerializableInvokeResult {
     Failed(SerializableError),
     Pending,
     Completed(Result<WitValue, RpcError>),
 }
+
+impl PartialEq for SerializableInvokeResult {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Failed(e1), Self::Failed(e2)) => e1 == e2,
+            (Self::Pending, Self::Pending) => true,
+            (Self::Completed(Ok(r1)), Self::Completed(Ok(r2))) => {
+                let v1: Value = r1.clone().into();
```

Comment:
is clone unavoidable here? maybe *r1 == *r2 or some other workaround?

Reply:
No idea did not try to optimize this code :)